src/tribler/core/utilities/async_interruption.py

```diff
@@ -0,0 +1,58 @@
+import asyncio
+
+
+def interruption(func):
+    """Decorator to allow interruption of a coroutine.
+
+    For more information, see the example at the end of this file.
+     Also check this: https://stackoverflow.com/questions/59586879/does-await-in-python-yield-to-the-event-loop
+    """
+
+    async def wrapper(*args, **kwargs):
```

Comment:
Can you also please apply [`wraps`](https://stackoverflow.com/questions/308999/what-does-functools-wraps-do) decorator to the wrapper function?
```suggestion
    @functools.wraps(func)
    async def wrapper(*args, **kwargs):
```